notes/trees.md

```diff
@@ -0,0 +1,234 @@
+# Trees.scala
+
+## Top level (`meta.Source`, `meta.Pkg`, `meta.Pkg`)
+
+```scala
+"package a".parse[Source].get.structure
+```
+
+## Declarations (`meta.Decl`)
+
+```scala
+"val a: Int".parse[Stat].get.structure
+```
+
+Decl            | Examples
+----------------|-----------
+Decl.Val        | val a
+Decl.Var        | var a
+Decl.Def        | def f: Int
+Decl.Type       | type T
+
+
+## Definitions (`meta.Defn`)
+
+```scala
+"val a = 1".parse[Stat].get.structure
+```
+
+Defn        | Examples
+------------|-----------
+Defn.Val    | val a = 1
+Defn.Var    | var a = 1
+Defn.Def    | def f = 1
+Defn.Macro  | def f = macro impl
+Defn.Type   | type T = Int
+Defn.Class  | class A
```

Comment:
can we use `class T` and `trait T` but `object A`? Classes and traits create types while objects create terms.

Reply:
T is for type parameters.